### PR TITLE
Improve message of Assertions.assertDoesNotThrow

### PR DESCRIPTION
**Pull request by olcbean**
If an unexpected error is thrown, the throwable message is added to the resulting `AssertionFailedError`.

Issue: #1703

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [ ] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertDoesNotThrow.java [hunk 1]
```diff
@@ -49,7 +50,7 @@ private static void assertDoesNotThrow(Executable executable, Object messageOrSu
 		}
 		catch (Throwable t) {
 			String message = buildPrefix(nullSafeGet(messageOrSupplier)) + "Unexpected exception thrown: "
```
Comment (sormuras) [inline junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertDoesNotThrow.java]: This line `String message = buildPrefix(nullSafeGet(messageOrSupplier)) + "Unexpected exception thrown: "` is duplicated below. Merge both into a new private `createMessage(messageOrSupplier, t)` or even a `createAssertionFailedError(messageOrSupplier, t)`?

**Comment by olcbean** [comment]
@sormuras thanks for the feedback! I just cleaned up the duplication. Can you have another look?

**Comment by sormuras** [comment]
Thank you @olcbean -- much appreciated.
